### PR TITLE
Update dependencies and stabilize tests

### PR DESCRIPTION
**Pull request by jglick**
Continuing #203 & #205, trying to stabilize the test suite here which is apparently loaded with cases of background activity of unclear timing.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,10 +28,12 @@
         <revision>1.49</revision>
         <changelist>999999-SNAPSHOT</changelist>
         <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-        <jenkins.version>2.361.4</jenkins.version>
+        <jenkins.version>2.387.3</jenkins.version>
         <useBeta>true</useBeta>
         <spotbugs.effort>Max</spotbugs.effort>
         <spotbugs.threshold>Low</spotbugs.threshold>
+        <!-- TODO pending inclusion in parent: -->
+        <jenkins-test-harness.version>2061.v04b_f07857801</jenkins-test-harness.version> <!-- TODO https://github.com/jenkinsci/jenkins-test-harness/pull/643 -->
```
Comment (jglick) [inline pom.xml:36]: https://github.com/jenkinsci/jenkins-test-harness/pull/643

File: src/test/java/hudson/plugins/copyartifact/CopyArtifactTest.java [hunk 2]
```diff
@@ -333,7 +333,7 @@ public void testCopyAllWithFingerprints() throws Exception {
     public void testCopyAllWithoutFingerprints() throws Exception {
         FreeStyleProject other = createArtifactProject(),
                          p = createProject(other.getName(), null, "", "", false, false, false, false);
-        FreeStyleBuild s = rule.assertBuildStatusSuccess(other.scheduleBuild2(0, new Cause.UserIdCause()).get());
+        FreeStyleBuild s = rule.buildAndAssertSuccess(other);
         FreeStyleBuild b = p.scheduleBuild2(0, new Cause.UserIdCause()).get();
         rule.assertBuildStatusSuccess(b);
```
Comment (jglick) [inline src/test/java/hudson/plugins/copyartifact/CopyArtifactTest.java:338]: Leaving more complex multiline idioms for proper tooling: https://github.com/openrewrite/rewrite-jenkins/issues/43

**Review by jglick — COMMENTED** [review]
I also noticed this plugin has several CRNL source files, which makes diffs weird.

**Review by MarkEWaite — APPROVED** [review]
Thanks!

**Comment by basil** [comment]
Is `org.jvnet.hudson.test.RemainingActivityListener.wait` not just a temporary workaround though?

**Comment by jglick** [comment]
Until someone finds the time to go through the tests in this plugin and clean them up.

**Comment by basil** [comment]
Mmm, "someone."

**Comment by basil** [comment]
I would personally not have approved this PR.

File: pom.xml [hunk 3]
```diff
@@ -28,10 +28,12 @@
         <revision>1.49</revision>
         <changelist>999999-SNAPSHOT</changelist>
         <gitHubRepo>jenkinsci/${project.artifactId}-plugin</gitHubRepo>
-        <jenkins.version>2.361.4</jenkins.version>
+        <jenkins.version>2.387.3</jenkins.version>
         <useBeta>true</useBeta>
         <spotbugs.effort>Max</spotbugs.effort>
         <spotbugs.threshold>Low</spotbugs.threshold>
+        <!-- TODO pending inclusion in parent: -->
+        <jenkins-test-harness.version>2062.v3efc79721e45</jenkins-test-harness.version>
```
Comment (basil) [inline pom.xml:36]: #226

**Pull request merged**